arelle/ModelValue.py

```diff
@@ -484,124 +610,127 @@ def time(value, castException=None):
     return Time(int(match.group(1)),int(match.group(2)),int(match.group(3)),ms,tzinfo(match.group(5)))
 
 class Time(datetime.time):
-    def __new__(cls, hour=0, minute=0, second=0, microsecond=0, tzinfo=None):
+
+    hour24: bool
+
+    def __new__(cls, hour: int = 0, minute: int = 0, second: int = 0, microsecond: int = 0, tzinfo: datetime.timezone | None = None) -> Time:
         hour24 = (hour == 24 and minute == 0 and second == 0 and microsecond == 0)
         if hour24: hour = 0
         time = datetime.time.__new__(cls, hour, minute, second, microsecond, tzinfo)
         time.hour24 = hour24
         return time
 
 class gYearMonth():
-    def __init__(self, year, month):
+    def __init__(self, year: int, month: int) -> None:
         self.year = int(year) # may be negative
         self.month = int(month)
-    def __repr__(self):
+    def __repr__(self) -> str:
         return self.__str__()
-    def __str__(self):
+    def __str__(self) -> str:
         return "{0:0{2}}-{1:02}".format(self.year, self.month, 5 if self.year < 0 else 4) # may be negative
-    def __eq__(self,other):
+    def __eq__(self,other: gYearMonth) -> bool: # type: ignore[override]
```

Comment:
Here (`__eq__` and `__ne__`) it seems I'm violating the Liskov substitution principle by typing like htis. It helps with the typing though, thus the ignore. Please advise if you find this to be a bad idea.

Reply:
Yeah, this type is incorrect. `__eq__` can be called with anything (and should handle other types). It's poorly implemented though (it should use `isinstance` and `NotImplemented` to [facilitate comparison fallbacks](https://docs.python.org/3/library/constants.html#NotImplemented)). You can update these (`isinstance` will give you the correct types) or just use `Any` for the param if you don't want to touch the implementation.
```python
    def __eq__(self, other: Any) -> bool:
        if not isinstance(other, gYearMonth):
            return NotImplemented
        return self.year == other.year and self.month == other.month
```

```python
    def __lt__(self, other: Any) -> bool:
        if not isinstance(other, gYearMonth):
            return NotImplemented
        return (self.year < other.year) or (self.year == other.year and self.month < other.month)
```

 etc.

Reply:
Also feel free to use the annotation thats on qname and delete a bunch of the unneeded implementations i.e. only need eq and lt and the annotation implicitly generates the rest

Reply:
> Also feel free to use the annotation thats on qname and delete a bunch of the unneeded implementations i.e. only need eq and lt and the annotation implicitly generates the rest

I don't feel confident enough to start deleting from these classes. Just adding type hints for now.